### DISCUSSION TITLE
Elastalert <> Slack Integration

### DISCUSSION THREAD

**Discussion by kennysden** [Security-Onion-Solutions/securityonion#2032] [Unsupported Versions] [answered]
I'm trying to set up elastalert to send to my slack webhook, but I'm getting this error when I test it rules

`OCI runtime exec failed: exec failed: container_linux.go:349: starting container process caused "exec: \"bash\": executable file not found in $PATH": unknown`

Has anyone set it up in slack before, how is it done? Thanks

**Comment by jertel** [comment]
What is the command line you are using to test the rules?

**Comment by stijnpieters** [comment, marked as answer]
so-elastalert-test tries to run 
`docker exec -it so-elastalert bash -c "elastalert-test-rule $RULE_NAME $OPTIONS" > $FILE_SAVE_LOCATION`
The so-elastalert container does not have a bash binary, only sh is present as far as I know.

2 easy solutions are available:
The first would be to edit /usr/sbin/so-elastalert-test and change the line above to use "sh" instead of "bash".
The second would be to manually enter the above command, but replacing bash with sh. 
(or you could 'docker exec -it so-elastalert sh', but that wastes a (almost) good script provisioned by security-onion)

**Reply by jertel** [reply]
Thanks for bringing this to our attention. I've created issue https://github.com/Security-Onion-Solutions/securityonion/issues/2048 for this. It will be resolved in the next planned release.
